palette/src/rgb/rgb.rs

```diff
@@ -953,14 +955,92 @@ where
     }
 }
 
+#[derive(Debug)]
+pub enum FromHexError {
+    ParseIntError(ParseIntError),
+    HexFormatError(&'static str)
+}
+
+impl From<ParseIntError> for FromHexError {
+    fn from(err: ParseIntError) -> FromHexError {
+        FromHexError::ParseIntError(err)
+    }
+}
+
+impl From<&'static str> for FromHexError {
+    fn from(err: &'static str) -> FromHexError {
+        FromHexError::HexFormatError(err)
+    }
+}
+impl core::fmt::Display for FromHexError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &*self {
+            FromHexError::ParseIntError(e) => {
+                write!(f, "ParseIntError: {}", e)
+            }
+            FromHexError::HexFormatError(s) => {
+                write!(f, "HexFormatError: {}, please use format '#fff', 'fff', '#ffffff' or\
+                'ffffff'.", s)
```

Comment:
I think it would be better to skip the `ParseIntError:` and `HexFormatError:` prefixes. I don't think they will be that interesting to someone who would read the error message, to be honest. They just want to know what went wrong and how to fix it. :slightly_smiling_face: